include/ck/tensor_operation/gpu/block/blockwise_softmax.hpp

```diff
@@ -108,6 +108,41 @@ struct BlockwiseSoftmax
         });
     }
 
+    template <typename CThreadBuffer>
+    __host__ __device__ void ApplyDropout(CThreadBuffer& in_thread_buf,
+                                          ushort p_dropout_16bits,
+                                          ck::philox ph,
```

Comment:
why is ApplyDropout a function of BlockwiseSoftmax?

Reply:
Because dropout is did on the result of softmax. Some of the parameters of softmax can be reused in dropout, such as MRepeat and KRepeat in this function. With these two parameter, we can calculate how many elements are on a thread, and calculate the offset of an single element.